From: SeongJae Park <sj@kernel.org>
Date: Fri, 4 Jul 2025 11:36:01 -0700
Subject: [PATCH] mm/damon/core: do not call ops.cleanup() when destroying
 targets

damon_operations.cleanup() is documented to be called for kdamond
termination, but also being called for targets destruction, which is
done for any damon_ctx destruction.  Nobody is using the callback for
now, though.  Remove the cleanup() call under the destruction.

Signed-off-by: SeongJae Park <sj@kernel.org>
---
 mm/damon/core.c | 5 -----
 1 file changed, 5 deletions(-)

diff --git a/mm/damon/core.c b/mm/damon/core.c
index 3c0c3cef43e7..3743511ed264 100644
--- a/mm/damon/core.c
+++ b/mm/damon/core.c
@@ -550,11 +550,6 @@ static void damon_destroy_targets(struct damon_ctx *ctx)
 {
 	struct damon_target *t, *next_t;
 
-	if (ctx->ops.cleanup) {
-		ctx->ops.cleanup(ctx);
-		return;
-	}
-
 	damon_for_each_target_safe(t, next_t, ctx)
 		damon_destroy_target(t);
 }
-- 
2.39.5

